### PR TITLE
apiextensions: add integration test which uncovers missing conversion on etcd read

### PR DESCRIPTION
**Pull request by sttts**
Test uncovering https://github.com/kubernetes/kubernetes/issues/68197.

### CODE REVIEW

**Comment by yue9944882** [comment]
sadly, this test is a flaky one. 😢everytime i ran it locally, it'll fail with an error `context canceled` or `grpc: the client connection is closing`.

**Comment by yue9944882** [comment]
are we hitting this issue #66477?

File: staging/src/k8s.io/apiextensions-apiserver/test/integration/versioning_test.go [hunk 1]
```diff
@@ -17,14 +17,98 @@ limitations under the License.
...
+			},
+		}
+	})
+	assert.NoError(t, err)
+
+	// patch until the apiVersion validation fails, i.e. the patch against v1beta1 is not valid.
+	// Note: if there is no conversion going on on etcd read, it's always the storage version we apply the patch against.
+	//       This is wrong of course and the following loop would time out because there is never an Invalid error returned.
+	t.Logf("Waiting for patch of non-storage version to fail")
+	i := 0
```
Comment (yue9944882) [inline staging/src/k8s.io/apiextensions-apiserver/test/integration/versioning_test.go]: xref: #68508

a `time.Sleep(time.Second)` will resolve the status error

**Comment by liggitt** [comment]
folded into https://github.com/kubernetes/kubernetes/pull/70087

/close

**Pull request closed without merging**